modules/navigation/3d/nav_mesh_queries_3d.cpp

```diff
@@ -295,6 +314,40 @@ void NavMeshQueries3D::_query_task_build_path_corridor(NavMeshPathQueryTask3D &p
 
 				// Only consider the connection to another polygon if this polygon is in a region with compatible layers.
 				const NavBaseIteration *owner = connection.polygon->owner;
+				bool skip_connection = false;
+				if (p_query_task.exclude_regions || p_query_task.include_regions) {
+					switch (owner->get_type()) {
+						case NavigationUtilities::PathSegmentType::PATH_SEGMENT_TYPE_REGION: {
+							if (p_query_task.exclude_regions && p_query_task.excluded_regions.has(owner->get_self())) {
+								skip_connection = true;
+							}
+							if (p_query_task.include_regions && !p_query_task.included_regions.has(owner->get_self())) {
```

Comment:
that could be an else if as skip_connection will already be true if we enter previous if